test/topology_experimental_raft/test_cdc_generation_clearing.py

```diff
@@ -63,13 +63,22 @@ async def tried_to_remove_new_gen() -> Optional[tuple[int, set[str], list[Host]]
     await check_system_topology_and_cdc_generations_v3_consistency(manager, hosts)
     second_gen_id = max(gen_ids)
 
-    await inject_error_one_shot(manager.api, servers[0].ip_addr, "clean_obsolete_cdc_generations_change_ts_ub")
-
-    logger.info("Bootstrapping third node")
-    servers += [await manager.server_add()]
-
-    # The first and second generations should be removed thanks to the above injection.
-    mark, gen_ids, hosts = await wait_for(tried_to_remove_new_gen, time.time() + 60)
-    logger.info(f"Generations after third clearing attempt: {gen_ids}")
-    assert len(gen_ids) == 1 and first_gen_id not in gen_ids and second_gen_id not in gen_ids
-    await check_system_topology_and_cdc_generations_v3_consistency(manager, hosts)
+    async with inject_error(manager.api, servers[0].ip_addr, "clean_obsolete_cdc_generations_change_ts_ub"):
```

Comment:
Does this hardened test actually reproduce the bug https://github.com/scylladb/scylladb/issues/16916?

Reply:
Any test that expects two clearings would fail with this bug in the code. After the first clearing, CDC generation publisher got stuck due to an error. The old version of this test performed only one clearing and it was the last step of the test. So, the test passed even though the clearing didn't work properly.

Suppose this PR didn't exist and we didn't have to change the clearing logic. Then, I would send a direct fix for #16916 and harden the test in a similar way -- by expecting two clearings. I did exactly that on my local machine while investigating this issue. The hardened version passed only after the direct fix.

However, we have this PR and it changes the clearing logic. The bugged code is no longer here and the test is adjusted. Nevertheless, #16916 showed that the test should be stronger by expecting at least two clearings instead of one because the first one may do something wrong without being noticed. So, I've decided to harden the test anyway.